src/objects/creep_shared.rs

```diff
@@ -262,14 +262,79 @@ pub unsafe trait SharedCreepProperties: RoomObjectProperties {
 unsafe impl SharedCreepProperties for Creep {}
 unsafe impl SharedCreepProperties for PowerCreep {}
 
+#[derive(Serialize, Deserialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub enum LineStyle {
+    Undefined,
+    Dashed,
+    Dotted,
+}
+js_serializable!(LineStyle);
+impl ToString for LineStyle {
+    fn to_string(&self) -> String {
+        match self {
+            LineStyle::Undefined => "undefined".to_owned(),
+            LineStyle::Dashed => "dashed".to_owned(),
+            LineStyle::Dotted => "dotted".to_owned(),
+        }
+    }
+}
+#[derive(Serialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct PolyStyle {
+    fill: String,
+    opacity: f64,
+    stroke: String,
+    stroke_width: f64,
+    line_style: LineStyle,
+}
+js_serializable!(PolyStyle);
+
+impl Default for PolyStyle {
+    fn default() -> Self {
+        PolyStyle {
+            fill: "#ffffff".to_string(),
+            opacity: 0.0,
+            stroke: "undefined".to_string(),
+            stroke_width: 0.1,
+            line_style: LineStyle::Undefined,
+        }
+    }
+}
+impl PolyStyle {
+    pub fn fill(mut self, fill: String) -> Self {
```

Comment:
It'd be nice to have a short doc comment on each of these methods - I think we can just copy the descriptions from https://docs.screeps.com/api/#RoomVisual.poly.